src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -51635,26 +51676,20 @@ void M2(C c1, bool b, object? x)
     void M3(C c1, bool b, object? x, object? y)
     {
         _ = (bool?)((y = 0) is 0 && c1.M(x = 0)) ?? false
-            ? x.ToString() + y.ToString() // 3
-            : x.ToString() + y.ToString(); // 4
+            ? x.ToString() + y.ToString() // 2
+            : x.ToString() + y.ToString();
     }
 
     bool M(object obj) { return true; }
 }
 ";
             // Note that we unsplit any conditional state after visiting the left side of `??`.
```

Comment:
```suggestion
```

I think this comment was wrong even before your change. It was probably the conversion to `bool?` within the LHS, not the LHS itself, which was causing an unsplit.